### ISSUE TITLE
Debian 11 Static VM - fdisk

### ISSUE THREAD

**Issue by zmeyers-ampere** [GoogleCloudPlatform/PerfKitBenchmarker#3667]
## Background
I'm using PKB to run workloads on a static GCP VM and on GCP VMs provisioned by PKB.

Both the static and provisioned VMs are using Debian 11. 

## Issue
There seems to be an issue with `fdisk` not being installed on Debian 11. PKB accounts for this in `perfkitbenchmarker/linux_virtual_machine.py` with a workaround, but this fix is not extended to `perfkitbenchmarker/static_virtual_machine.py` 

### Provisioned VMs
- Specifying `os_type: debian11` in my config works just fine, PKB works as expected: spins up/tears down the VM and runs the workloads successfully

### Static VMs
- If I don't specify `os_type` in my config, PKB defaults to `ubuntu1804` and then fails on the usage of `fdisk` (see details below)
```
2022-07-27 18:29:09,139 6f3d18c4 MainThread coremark(1/1) ERROR    Exception running benchmark
Traceback (most recent call last):
  File "/perfkitbenchmarker/pkb.py", line 1299, in RunBenchmarkTask
    RunBenchmark(spec_for_run, collector)
  File "/perfkitbenchmarker/pkb.py", line 1078, in RunBenchmark
    DoProvisionPhase(spec, detailed_timer)
  File "/perfkitbenchmarker/pkb.py", line 755, in DoProvisionPhase
    spec.Provision()
  File "/perfkitbenchmarker/benchmark_spec.py", line 718, in Provision
    vm_util.RunThreaded(self.PrepareVmAfterBoot, self.vms)
  File "/perfkitbenchmarker/background_tasks.py", line 647, in RunThreaded
    return RunParallelThreads(target_arg_tuples,
  File "/perfkitbenchmarker/background_tasks.py", line 582, in RunParallelThreads
    return _RunParallelTasks(
  File "/perfkitbenchmarker/background_tasks.py", line 556, in _RunParallelTasks
    raise parallel_exception_class(
perfkitbenchmarker.errors.VmUtil.ThreadException: The following exceptions occurred during parallel execution:
Exception occurred while calling PrepareVmAfterBoot(34.68.0.208):
Traceback (most recent call last):
  File "/perfkitbenchmarker/background_tasks.py", line 252, in Run
    self.return_value = self.target(*self.args, **self.kwargs)
  File "/perfkitbenchmarker/benchmark_spec.py", line 977, in PrepareVmAfterBoot
    vm.PrepareVMEnvironment()
  File "/perfkitbenchmarker/linux_virtual_machine.py", line 471, in PrepareVMEnvironment
    self.RecordAdditionalMetadata()
  File "/perfkitbenchmarker/linux_virtual_machine.py", line 783, in RecordAdditionalMetadata
    self.os_metadata.update(self.partition_table)
  File "/perfkitbenchmarker/linux_virtual_machine.py", line 740, in partition_table
    partition_tables = self.RemoteCommand(cmd)[0]
  File "/perfkitbenchmarker/linux_virtual_machine.py", line 965, in RemoteCommand
    return self.RemoteCommandWithReturnCode(*args, **kwargs)[:2]
  File "/perfkitbenchmarker/linux_virtual_machine.py", line 981, in RemoteCommandWithReturnCode
    return self.RemoteHostCommandWithReturnCode(*args, **kwargs)
  File "/perfkitbenchmarker/linux_virtual_machine.py", line 1053, in RemoteHostCommandWithReturnCode
    raise errors.VirtualMachine.RemoteCommandError(error_text)
perfkitbenchmarker.errors.VirtualMachine.RemoteCommandError: Got non-zero return code (1) executing sudo fdisk -l
Full command: ssh -A -p 22 <user>@<ip_address> -2 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -o PreferredAuthentications=publickey -o PasswordAuthentication=no -o ConnectTimeout=5 -o GSSAPIAuthentication=no -o ServerAliveInterval=30 -o ServerAliveCountMax=10 -i <ssh_key_path> -o ControlPath="/tmp/perfkitbenchmarker/runs/6f3d18c4/ssh/%h" -o ControlMaster=auto -o ControlPersist=30m sudo fdisk -l
STDOUT: STDERR: sudo: fdisk: command not found
```
- Specifying `os_type: debian11` in my config raises a SubClassNotFoundError (details below)
```
2022-07-27 18:11:02,810 20ddbd71 MainThread coremark(1/1) ERROR    Exception running benchmark
Traceback (most recent call last):
  File "/perfkitbenchmarker/pkb.py", line 1299, in RunBenchmarkTask
    RunBenchmark(spec_for_run, collector)
  File "/perfkitbenchmarker/pkb.py", line 1078, in RunBenchmark
    DoProvisionPhase(spec, detailed_timer)
  File "/perfkitbenchmarker/pkb.py", line 734, in DoProvisionPhase
    spec.ConstructVirtualMachines()
  File "/perfkitbenchmarker/benchmark_spec.py", line 541, in ConstructVirtualMachines
    vms = self.ConstructVirtualMachineGroup(group_name, group_spec)
  File "/perfkitbenchmarker/benchmark_spec.py", line 438, in ConstructVirtualMachineGroup
    static_vm_class = static_vm.GetStaticVmClass(vm_spec.os_type)
  File "/perfkitbenchmarker/static_virtual_machine.py", line 314, in GetStaticVmClass
    return resource.GetResourceClass(virtual_machine.BaseVirtualMachine,
  File "/perfkitbenchmarker/resource.py", line 49, in GetResourceClass
    raise errors.Resource.SubclassNotFoundError(
perfkitbenchmarker.errors.Resource.SubclassNotFoundError: No BaseVirtualMachine subclass defined with the attributes: {'CLOUD': 'Static', 'OS_TYPE': 'debian11'}
```

## Proposed Solution
To workaround this issue I can `sudo apt install fdisk` on the static VM.

However, I think a more permanent fix in the codebase to support the debian11 mixin for static virtual machines would be better for anyone else who may encounter this issue.

**Comment by bvliu** [comment]
Supporting a debian11 mixin for static VMs seems like a good solution. Feel free to send us a PR.

**Comment by zmeyers-ampere** [comment]
Sounds good, I'll get working on that, thank you for the response.

**Comment by pmkc** [comment]
Fixed in #3761
